### PR TITLE
TGUI maintenance chores and juke upgrades

### PR DESCRIPTION
**Pull request by Crossedfall**
## About The Pull Request

Ports several juke/tgs changes to resolve some of our technical debt and to get the TGUI dev server working again.

### Ported the following:
* https://github.com/tgstation/tgstation/pull/60859
* https://github.com/tgstation/tgstation/pull/61025
* https://github.com/tgstation/tgstation/pull/61005
* https://github.com/tgstation/tgstation/pull/61081
* https://github.com/tgstation/tgstation/pull/61110
* https://github.com/tgstation/tgstation/pull/62067
* https://github.com/tgstation/tgstation/pull/61399

## Why It's Good For The Game

General TGUI improvements and QoL changes for development. 

## ⚠️ IMPORTANT CHANGES ⚠️ 

Interacting with TGUI has changed drastically. There is no longer a `bin/tgui` script to interact with. Instead, developers will need to use either the windows scripts found in the `bin/` root directory, via Juke, or via yarn. Excerpt from the updated `tgui/README.md` below.

<details>
<summary>TGUI usage</summary>

**Via provided cmd scripts (Windows)**:

- `bin/tgui-build` - Build tgui in production mode and run a full suite of code checks.
- `bin/tgui-dev` - Launch a development server.
  - `bin/tgui-dev --reload` - Reload byond cache once.
  - `bin/tgui-dev --debug` - Run server with debug logging enabled.
  - `bin/tgui-dev --no-hot` - Disable hot module replacement (helps when doing development on IE8).
- `bin/tgui-sonar` - Analyze code with SonarQube.
- `bin/tgui-bench` - Run benchmarks.

> To open a CMD or PowerShell window in any open folder, right click **while holding Shift** on any free space in the folder, then click on either `Open command window here` or `Open PowerShell window here`.

**Via Juke Build (cross-platform)**:

- `tools/build/build tgui` - Build tgui in production mode.
- `tools/build/build tgui-dev` - Build tgui in production mode.
  - `tools/build/build tgui-dev --reload` - Reload byond cache once.
  - `tools/build/build tgui-dev --debug` - Run server with debug logging enabled.
  - `tools/build/build tgui-dev --no-hot` - Disable hot module replacement (helps when doing development on IE8).
- `tools/build/build tgui-lint` - Show (and auto-fix) problems with the code.
- `tools/build/build tgui-sonar` - Analyze code with SonarQube.
- `tools/build/build tgui-test` - Run unit and integration tests.
- `tools/build/build tgui-analyze` - Run a bundle analyzer.
- `tools/build/build tgui-bench` - Run benchmarks.
- `tools/build/build tgui-clean` - Clean up tgui folder.

> With Juke Build, you can run multiple targets together, e.g.:
> ```
> tools/build/build tgui tgui-lint tgui-tsc tgui-test
> ```

**Via Yarn (cross-platform)**:

Run `yarn install` once to install tgui dependencies.

- `yarn tgui:build` - Build tgui in production mode.
  - `yarn tgui:build [options]` - Build tgui with custom webpack options.
- `yarn tgui:dev` - Launch a development server.
  - `yarn tgui:dev --reload` - Reload byond cache once.
  - `yarn tgui:dev --debug` - Run server with debug logging enabled.
  - `yarn tgui:dev --no-hot` - Disable hot module replacement (helps when doing development on IE8).
- `yarn tgui:lint` - Show (and auto-fix) problems with the code.
- `yarn tgui:sonar` - Analyze code with SonarQube.
- `yarn tgui:tsc` - Check code with TypeScript compiler.
- `yarn tgui:test` - Run unit and integration tests.
- `yarn tgui:analyze` - Run a bundle analyzer.
- `yarn tgui:bench` - Run benchmarks.
</details>

## Testing Photographs and Procedure

- Built the base using the default VSC command (Ctrl+Shift+B)
- Built the base using the debug function (F5)
- Built the base using `BUILD.bat`
- Built the base using `tools/build/build`
- Used all of the `bin/*.cmd` scripts
- Verified that the TGUI-dev server worked (see screenshots)
- Tested basic TGUI elements and windows

<details>
<summary>Screenshots&Videos</summary>

image

image

image

</details>

## Changelog
:cl:stylemistake,SplinterGP,mokulus
code: Treat DreamMaker warnings as errors in CI
refactor: Performance of core tgui components was improved (Button, Icon, Flex, Stack). UI should feel a tiny bit faster.
/:cl:

### CODE REVIEW

**Comment by Crossedfall** [comment]
Passed bare-minimum testing on Staging. Queued for an extended TM on Sage.

File: tgui/packages/tgui/interfaces/TechFab.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { Stack, Collapsible, Tooltip, Icon, Box, Button, LabeledList, Input, Sec
 import { Window } from '../layouts';
 import { Fragment } from 'inferno';
 import { capitalize, createSearch } from 'common/string';
-import { sendLogEntry } from 'tgui-dev-server/link/client';
+import { sendLogEntry } from 'tgui-dev-server/link/client.cjs';
```
Comment (KubeRoot) [inline tgui/packages/tgui/interfaces/TechFab.js:6]: This should probably be removed, seems unused, looks like a leftover from testing.

**Review by KubeRoot — APPROVED** [review]
Looks good, at least the stuff that doesn't go over my head, except for one nitpick.

File: tgui/packages/tgui/interfaces/TechFab.js (hunk 1 above)
Comment (KubeRoot) [reply tgui/packages/tgui/interfaces/TechFab.js:6]: Leaving this for potential cleanup for later

Comment (Crossedfall) [reply tgui/packages/tgui/interfaces/TechFab.js:6]: There are a couple of those from when Frost originally ported it all. There are also some other unused imports here and there, but eh it doesn't really hurt anything to be there. We could probably do a cleanup pass later, but that should be separate I think

**Pull request merged**